### ISSUE TITLE
Couldn't enable `opam-switch-mode` in Coq projects

### ISSUE THREAD

**Issue by lzy0505** [doomemacs/doomemacs#8085]
### I confirm that...

- [X] I have searched the [issue tracker](https://github.com/doomemacs/doomemacs/issues), [documentation](https://docs.doomemacs.org), [FAQ](https://docs.doomemacs.org/-/faq), [Discourse](https://discourse.doomemacs.org), and [Google](https://google.com), in case this issue has already been reported/resolved.

- [X] I have read "[How to Debug Issues](https://doomemacs.org/d/how2debug)", and will use it to provide as much information about this issue as possible.

- [X] The issue can be reproduced on the **latest** available commit of Doom.

- [X] The issue can be reproduced on a stable release of Emacs, such as 27, 28, or 29. *(Unstable versions end in .50, .60, or .9x)*

### Expected behavior

When I enable `opam-switch-mode` in a Coq project, there shouldn't be any errors.

### Current behavior

I got `opam-switch-set-switch: Symbol’s function definition is void: tuareg-opam-current-compiler`.

This bug seems to have been introduced after #7806, which calls this function when first loading `opam-switch-mode`, regardless of whether the `tuareg` package (the OCaml mode) is loaded.

While opam is commonly used for OCaml projects, it is also used in some Coq projects. 

### Steps to reproduce

1. Enable `coq` module `config.el`
2. Open a `.v` file in a Coq project
3. `M-x opam-switch-mode`

### System Information

https://pastebin.com/fEe1nJcK

**Comment by LukeXuan** [comment]
I just encountered the same problem. It's caused by the configuration in `modules/lang/ocaml/config.el` (assuming you also enabled ocaml in `init.el`): the `use-package!` calls `tuareg-opam-curent-compiler`. An easy fix would be to remove that line, as I don't see a purpose there.

**Comment by zeeshanlakhani** [comment]
confirming this issue as well.

**Comment by lzy0505** [comment]
#8132 might fix it.

**Comment by zeeshanlakhani** [comment]
> [#8132](https://github.com/doomemacs/doomemacs/pull/8132) might fix it.

Yeah, I tried overriding the opam-switch-mode `use-package!` from that PR to no avail. I'll wait and see on it for sure.

**Comment by jeremyThibault** [comment]
Hi, indeed the aim of #8132 was to fix this issue. Note that the fix is in two parts: (1) [this commit](https://github.com/doomemacs/doomemacs/pull/8132/commits/bdb4d6064650a5435c5f153b9180aac82b7c5db5) wraps the call to `tuareg-opam-curent-compiler` in an `!after tuareg` block so that it's not called unless `tuareg` is loaded; and (2) [this other commit](https://github.com/doomemacs/doomemacs/pull/8132/commits/ca5f12314c5d11cdc174dc017af713def95091be) adds configuration for `opam-switch-mode` to the `coq` module.

It works for me, so if there's anything I can do to help please let me know. I don't know how to fix the issue without modifying the `ocaml` module though, so I don't know if it's possible just by modifying your `config.el` file.
